torch/testing/_internal/common_cuda.py

```diff
@@ -32,3 +35,41 @@ def initialize_cuda_context_rng():
         for i in range(torch.cuda.device_count()):
             torch.randn(1, device="cuda:{}".format(i))
         __cuda_ctx_rng_initialized = True
+
+
+def tf32_is_not_fp32():
+    if not torch.cuda.is_available():
+        return False
+    if torch.cuda.get_device_properties(torch.cuda.current_device()).major < 8:
+        return False
+    return True
+
+
+@contextlib.contextmanager
+def setup_tf32(dtype, rtol=0.001, atol=1e-5):
```

Comment:
I think we should stash the current value of `torch.backends.cuda.matmul.allow_tf32` here...

Reply:
done